docs/en/latest/plugins/ai-prompt-decorator.md

```diff
@@ -0,0 +1,112 @@
+  - Apache APISIX
+  - API Gateway
+  - Plugin
+  - ai-prompt-decorator
+description: This document contains information about the Apache APISIX ai-prompt-decorator Plugin.
+---
+
+<!--
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+## Description
+
+The `ai-prompt-decorator` plugin simplifies access to AI providers and models by appending or prepending
+prompts into the request.
+
+## Plugin Attributes
+
+| **Field**         | **Type** | **Description**                                     | **Required**    |
+| ----------------- | -------- | --------------------------------------------------- | --------------- |
+| `prepend`         | Array    | An array of prompt objects to be prepended          | Conditionally\* |
+| `prepend.role`    | String   | Role of the message (`system`, `user`, `assistant`) | Yes             |
+| `prepend.content` | String   | Content of the message. Minimum length: 1           | Yes             |
+| `append`          | Array    | An array of prompt objects to be appended           | Conditionally\* |
+| `append.role`     | String   | Role of the message (`system`, `user`, `assistant`) | Yes             |
+| `append.content`  | String   | Content of the message. Minimum length: 1           | Yes             |
+
+\* **Conditionally Required**: At least one of `prepend` or `append` must be provided.
+
+## Example usage
+
+Create a route with the `ai-prompt-decorator` plugin like so:
+
+```shell
+curl "http://127.0.0.1:9180/apisix/admin/routes/1" -X PUT \
+  -H "X-API-KEY: ${ADMIN_API_KEY}" \
+  -d '{
+    "uri": "/v1/chat/completions",
+    "plugins": {
+      "ai-prompt-decorator": {
+        "prepend":[
+          {
+            "role": "system",
+            "content": "I have exams tomorrow so explain conceptually and briefly"
+          }
+        ],
+        "append":[
+          {
+            "role": "system",
+            "content": "End the response with an analogy."
+          }
+        ]
+      }
+    },
+    "upstream": {
+      "type": "roundrobin",
+      "nodes": {
+        "api.openai.com:443": 1
+      },
+      "pass_host": "node",
+      "scheme": "https"
+    }
+  }'
+```
+
+The upstream node can be any arbitrary value because it won't be contacted.
```

Comment:
I understand your intention but maybe remove this sentence 🤔 reads a bit confusing

Reply:
this is a key information so we should keep it, no?

Reply:
whoops sorry this is a wrong info 😂. Good catch 🙏🏼